pkg/querier/store_gateway_client.go

```diff
@@ -49,6 +49,15 @@ func dialStoreGatewayClient(clientCfg grpcclient.Config, addr string, requestDur
 	}, nil
 }
 
+type zoneAwareStoreGatewayClient struct {
+	*storeGatewayClient
+	zone string
+}
+
+func (c *zoneAwareStoreGatewayClient) Zone() string {
+	return c.zone
+}
+
 type storeGatewayClient struct {
```

Comment:
Can the `Zone()` method be added to the existing client? By default it can take a value of `""`.

Reply:
This needs more changes on the pool client struct

Reply:
Changes reverted so should be fixed.